src/components/expression/expression-renderer.js

```diff
@@ -673,23 +895,68 @@ genotet.ExpressionRenderer.prototype.highlightLabelsForClickedCell_ = function(
 };
 
 /**
+ * Unhighlights the label of the clicked cells for the heatmap.
+ * @param {!genotet.ExpressionRenderer.Cell} cell
+ * @private
+ */
+genotet.ExpressionRenderer.prototype.highlightLabelsForClickedCell_ =
+  function(cell) {
+    this.svgGeneLabels_
+      .selectAll('text')
+      .classed('click-highlighted', function(d, i) {
+        return cell.row == i;
+      });
+    this.svgConditionLabels_
+      .selectAll('text')
+      .classed('click-highlighted', function(d, i) {
+        return cell.column == i;
+      });
+  };
+
+/**
+ * Unhighlights all the labels of cells for the heatmap.
+ * @private
+ */
+genotet.ExpressionRenderer.prototype.unhighlightLabelsForClickedCell_ =
+  function() {
+    this.svgGeneLabels_
+      .selectAll('text')
+      .classed('click-highlighted', false);
+    this.svgConditionLabels_
+      .selectAll('text')
+      .classed('click-highlighted', false);
+  };
+
+/**
  * Highlights the labels of clicked profile.
- * @param {*} profile TODO(liana): fill the type.
+ * @param {!genotet.ExpressionRenderer.Profile} profile
  * @private
  */
 genotet.ExpressionRenderer.prototype.highlightLabelsForClickedProfile_ =
-    function(profile) {
-  this.svgGeneLabels_.selectAll('text')
-    .attr('fill', 'black');
-  this.svgConditionLabels_.selectAll('text')
-    .attr('fill', 'black');
-  this.svgGeneLabels_.select('text:nth-child(' +
-      (profile.row + 1) + ')')
-    .attr('fill', profile.color);
-  this.svgConditionLabels_.select('text:nth-child(' +
-      (profile.hoverColumn + 1) + ')')
-    .attr('fill', profile.color);
-};
+  function(profile) {
+    this.unhighlightLabelsForClickedProfile_();
+    this.svgGeneLabels_.select('text:nth-child(' + (profile.row + 1) + ')')
+      .attr('fill', profile.color)
```

Comment:
I cannot add classes simply here because the color of the path need to be calculated. They are not constants.